drivers/flash/Kconfig

```diff
@@ -162,4 +162,15 @@ config SOC_FLASH_MCUX_DEV_NAME
 	help
 	  Specify the device name for the flash driver.
 
+config SOC_FLASH_WRITE_BLOCK_SIZE
+	int
+	prompt "write block size (fixed)" if SOC_FLASH_NRF5 || SOC_FLASH_STM32 ||\
+	SOC_FLASH_QMSI
+	range 1 1 if SOC_FLASH_NRF5
+	range 4 4 if SOC_FLASH_STM32
+	range 4 4 if SOC_FLASH_QMSI
+	depends on FLASH
+	help
+	 The minimum write block size supported by the SoC flash driver.
+
```

Comment:
where is this being used? The dependency should be done using depends on, and not on the prompt, or are you trying to do something else here?

Reply:
Nor `depends on FALSH && (SOC_FLASH_NRF5 || SOC_FLASH_STM32 || SOC_FLASH_QMS)` neither `depends on (SOC_FLASH_NRF5 || SOC_FLASH_STM32 || SOC_FLASH_QMS)` dose not work: config dispersing. For construction above this config is added in `flash hardware support menu` as expected.